libflux/flux-core/src/ast/tests.rs

```diff
@@ -2131,3 +2131,48 @@ fn test_ast_json_roundtrip() {
     };
     assert_eq!(ast, roundtrip_ast);
 }
+
+/// Locator makes constructing an SourceLocation from a string simple.
+pub struct Locator<'a> {
+    source: &'a str,
+    lines: Vec<u32>,
+}
+
+impl<'a> Locator<'a> {
+    /// Create a new Locator for the given source code
+    pub fn new(source: &'a str) -> Self {
+        let mut lines = Vec::new();
+        lines.push(0);
```

Comment:
This is to pad the lines, right? So the lines are one-indexed rather than zero-indexed?